ksvd/__init__.py

```diff
@@ -64,16 +64,29 @@ def _transform(self, D, X):
         return orthogonal_mp_gram(
             gram, Xy, n_nonzero_coefs=n_nonzero_coefs).T
 
-    def fit(self, X):
+    def fit(self, X, pretrained_D=None, verbose_log=False):
         """
         Parameters
         ----------
-        X: shape = [n_samples, n_features]
+        X:
+            shape = [n_samples, n_features]
+
+        pretrained_D:
+            Load pretrained dictionary
+
+        verbose_log:
+            Print current iteration and error
         """
-        D = self._initialize(X)
+        if pretrained_D is None:
+            D = self._initialize(X)
+        else:
+            D = pretrained_D
+
         for i in range(self.max_iter):
             gamma = self._transform(D, X)
             e = np.linalg.norm(X - gamma.dot(D))
+            if verbose_log:
+                print('Iter: %d, Err: %.4f' % (i, e))
```

Comment:
Could you use `logging.getLogger` like below? I would like to conform to the python logging ecosystem.
```
diff --git a/ksvd/__init__.py b/ksvd/__init__.py
index 72b433f..00a4056 100644
--- a/ksvd/__init__.py
+++ b/ksvd/__init__.py
@@ -2,6 +2,8 @@
 import numpy as np
 import scipy as sp
 from sklearn.linear_model import orthogonal_mp_gram
+from logging import getLogger
+logger = getLogger(__name__)


 class ApproximateKSVD(object):
@@ -85,8 +87,7 @@ class ApproximateKSVD(object):
         for i in range(self.max_iter):
             gamma = self._transform(D, X)
             e = np.linalg.norm(X - gamma.dot(D))
-            if verbose_log:
-                print('Iter: %d, Err: %.4f' % (i, e))
+            logger.info('Iter: %d, Err: %.4f' % (i, e))
             if e < self.tol:
                 break
             D, gamma = self._update_dict(X, D, gamma)
```

And, if you would like to print the log, please set `logging.basicConfig(level=logging.INFO)` on a your script.

Reply:
OK. I will create a new pull request later.